File: POTDS/SmallestPositiveMissingNumber.java
import java.util.HashMap;

public class SmallestPositiveMissingNumber {
    public static void main(String[] args) {
        int[] arr = {1,2,3,4,5};
        int ans = missingNumber(arr , arr.length);
        System.out.println(ans);
    }
    static int missingNumber(int arr[], int size)
    {
        HashMap<Integer , Integer> m = new HashMap<>();
        for(int i : arr)
        {
            if(i > 0)
            m.put(i , 1);
        }
        for(int i = 1 ; i <= size+1 ; i++)
        {
            if(!m.containsKey(i))
            {
                return i;
            }
        }
        return 1;
    }
}

File: POTDS/SmallestPositiveMissingNumber.cpp
#include <bits/stdc++.h>
using namespace std;

int missingNumber(int arr[], int n);

int main()
{
    int arr[5] = {1, 2, 3, 4, 5};
    int size = end(arr) - begin(arr);
    int ans = missingNumber(arr, size);
    cout << ans;
}

int missingNumber(int arr[], int n)
{
    int c = 1;
    sort(arr, arr + n);
    for (int i = 0; i < n; i++)
    {
        if (arr[i] == c)
            c++;
    }
    return c;
}
